src/main/docs/guide/generator/registration.adoc

```diff
@@ -0,0 +1,36 @@
+The build time generation is configured with the `BeanGeneratorTask`. The following code snippet shows an example configuration for generating schemas inside an input folder called `ref`:
+
+++++
+include::src/main/docs/guide/generator/BeanTaskRegistration.html[]
+++++
```

Comment:
what about Maven?

Reply:
Gonna add it in the maven plugin repository